services/app-api/src/postgres/contractAndRates/proto_to_db_Documents.ts

```diff
@@ -0,0 +1,184 @@
+                    })
+
+                results.contractSupportingDocuments.push(
+                    createdContractSupportDoc
+                )
+            }
+        }
+
+        // Handle Rate and Rate Supporting Documents
+        for (const rateInfo of formData.rateInfos) {
+            const rateRevision = await client.rateRevisionTable.findFirst({
+                where: {
+                    rateID: revision.pkgID,
+                    rateType: rateInfo.rateType,
+                    rateCapitationType: rateInfo.rateCapitationType,
+                    rateDateStart: rateInfo.rateDateStart,
+                    rateDateEnd: rateInfo.rateDateEnd,
+                },
+            })
+
+            if (rateRevision) {
+                for (const rateDoc of rateInfo.rateDocuments) {
+                    const existingRateDoc = await client.rateDocument.findFirst(
+                        {
+                            where: {
+                                sha256: rateDoc.sha256 ?? '',
+                                rateRevisionID: rateRevision.id,
+                            },
+                        }
+                    )
+
+                    if (!existingRateDoc) {
+                        const rateDocument: RateDocument = {
+                            id: uuid(),
+                            createdAt: revision.createdAt,
+                            updatedAt: new Date(),
+                            name: rateDoc.name,
+                            s3URL: rateDoc.s3URL,
+                            sha256: rateDoc.sha256 ?? '',
+                            rateRevisionID: rateRevision.id,
+                        }
+
+                        const createdRateDoc = await client.rateDocument.create(
+                            {
+                                data: rateDocument,
+                            }
+                        )
+
+                        results.rateDocuments.push(createdRateDoc)
+                    }
+                }
+
+                for (const supportRateDoc of rateInfo.supportingDocuments) {
+                    const existingRateSupportDoc =
+                        await client.rateSupportingDocument.findFirst({
+                            where: {
+                                sha256: supportRateDoc.sha256 ?? '',
+                                rateRevisionID: rateRevision.id,
+                            },
+                        })
+
+                    if (!existingRateSupportDoc) {
+                        const rateSupportDocument: RateSupportingDocument = {
+                            id: uuid(),
+                            createdAt: revision.createdAt,
+                            updatedAt: new Date(),
+                            name: supportRateDoc.name,
+                            s3URL: supportRateDoc.s3URL,
+                            sha256: supportRateDoc.sha256 ?? '',
+                            rateRevisionID: rateRevision.id,
```

Comment:
I believe this field is now required.

Reply:
Yeah I think you're right, but it looks like common-code's `healthPlanFormData` types still have it as an optional, for example:

```typescript
type SubmissionDocument = {
    id?: string
    name: string
    s3URL: string
    sha256?: string
    documentCategories: DocumentCategoryType[]
}
```

Is there any danger in updating that to required?

Reply:
> Is there any danger in updating that to required?
I don't think so since we migrated existing docs to have the sha and any new documents will generate a sha on upload. The only thing that worries me is if `calculateSHA256(file)` fails, we just set `sha256` to a empty string. Also possibility of a lot of fixing the code for the type change.

When I'm converting HPP rate type to our new domain rate type in `convertHealthPlanPackageRatesToDomain`, I'm generating a sha if there isn't one. This doesn't solve the empty string possibility though.

Reply:
Opened a ticket for this as follow on work: https://qmacbis.atlassian.net/browse/MCR-3527